docs/tutorials/zero-to-hero.md

```diff
@@ -308,601 +116,78 @@ export function getResponseByKey(key: string): string {
-
-To wrap up this tutorial, your mission, should you choose to accept it, is to create a simple smart contract that uses the price of bitcoin that we saved to the blockchain.
-
-The pseudo code is simple:
-
-```text
-btcPrice = price of btc
-if btcPrice >= 5000:
-  betOutcome = "Pay Moon Hodler 2000 USD"
-else:
-  betOutcome = "Pay FUD Hodler 2000 USD"
-
-save betOutcome to blockchain with key of 'betOutcome'
-```
-Although in this smart contract, we're just saving a string to the blockchain, you could imagine that we could instead replace that with logic which pays out a specific wallet address.
-
-As a final touch, build out a "finalize outcome" button on the front end that kicks off this logic, and then display the outcome on the screen.
-
-> Now you try it.
-> - Implement a new function in AssemblyScript on the smart contract that provides the expected behavior of the pseudo code above.  (hint: use the following signature)  \
-> `export function finalizeBet(): void { /* your implementation */ }`
-> - Surface the new contract method in JavaScript as before (hint: it's a `change` method)
-> - Implement a new function in JavaScript in that invokes `contract.finalizeBet()` as per the pseudo code above
-> - Add a button to invoke the method (and optionally render the result to the console or div)
-
-Hints:
-- string --> number `I32.parseInt(numberAsText)`
-- number --> string `someNumber.toString()`
-- at the time of writing, one should avoid using `i32()` and friends for typecasting strings to numbers
-
-
-How did it go?
-
-You can see how we did it here...
-
-> In the file `assembly/main.ts`
-> - We appended the code below to the bottom of the file
-
-```ts
-export function finalizeBet(): void {
-  let price = storage.getString("btc-price");
-
-  // expecting price as string in the format #,###.####
-  price = price.split(',').join('').split('.')[0];
-
-  if (price) {
-    let btcPrice: i32 = I32.parseInt(price);
-    let overUnderPrice: i32 = 6000;
-    let outcome: string;
-
-    if (btcPrice > overUnderPrice) {
-      outcome = "BTC price is " + btcPrice.toString() + " - Pay Moon Hodler 2000 USD"
-    } else {
-      outcome = "BTC price is " + btcPrice.toString() + " - Pay FUD Hodler 2000 USD"
-    }
-
-    storage.setString("betOutcome", outcome)
-
-  } else {
-    storage.setString("betOutcome", "btc price is undefined")
-  }
+  //stripping only the data we want from the API response
+  let data = body.bpi.USD.rate
+  //Saving the data to the blockchain by calling the Oracle Contracts setResponse function
+  await window.contract.setResponse({ apiResponse: data });
+  // Check to see if the data was saved properly
+  let apiResponse = await contract.getResponse();
+  console.log(`${apiResponse} is the API response available to the Oracle on-chain`);
 }
 ```
 
-> In the file `src/main.js`
-> - We surfaced the new contract method `finalizeBet` to our JavaScript contract
-
+Then replace the event handler of the button with the following:
```

Comment:
where? the say-hi that we were told to overwrite I take it?
again, let's explicitly tell them what they're replacing:

```
// Adding an event to a say-hi button.
  document.getElementById('say-hi').addEventListener('click', () => {
    // We call say Hi and then update who said Hi last.
    window.contract.sayHi().then(updateWhoSaidHi);
  });
```

Reply:
done